src/Text/PrettyPrint/Bernardy/Interface.idr

```diff
@@ -20,15 +20,74 @@ public export %inline
 prettyArg : Pretty a => {opts : _} -> a -> Doc opts
 prettyArg = prettyPrec App
 
+onLineAfter : {opts : _} -> Doc opts -> Doc opts -> Doc opts
+onLineAfter l d = vappend d (indent 2 l)
+
+-- multilineCon :
+--      {opts : _}
+--   -> (con, firstArg : Doc opts)
+--   -> (args : List (Doc opts))
+--   -> Doc opts
+-- multilineCon con fa as = foldl vcat fa as `onLineAfter` con
+
 ||| Utility for pretty printing Idris data types.
 |||
+||| Constructor name and arguments are layed out on a single line
+||| if and only if they fit the page width
+||| and all pretty printed arguments use only a single line. If
+||| one of these conditions does not hold, the list of arguments is
+||| indented by two spaces and layed out vertically after the constructor
+||| name.
+|||
 ||| ```idris example
 ||| prettyCon "Just" [prettyArg v]
 ||| ```
-public export
+export
 prettyCon : {opts : _} -> Prec -> String -> List (Doc opts) -> Doc opts
-prettyCon p s [] = line s
-prettyCon p s ds = parenthesise (p >= App) (hangSep 2 (line s) (sep ds))
+prettyCon p c []   = line c
```

Comment:
Is there any reason this takes a `Prec` but doesn't use it? (and `prettyRecord`)

Reply:
Thanks for spotting this. I forgot to add a call to `parenthesise`. This should be fixed now.